miner/worker.go

```diff
@@ -88,17 +89,19 @@ type environment struct {
 	tcount    int            // tx count in cycle
 	gasPool   *core.GasPool  // available gas used to pack transactions
 
-	header   *types.Header
-	txs      []*types.Transaction
-	receipts []*types.Receipt
+	header           *types.Header
+	txs              []*types.Transaction
+	receipts         []*types.Receipt
+	executionResults []*types.ExecutionResult
```

Comment:
I think having `executionResults` here is not enough. We will store extra trace here, e.g., access to accounts, storages, and code. We need to export extra information during the VM execution. So using `BlockResult` type here is more future proof.

Reply:
The `executionResults` is appended in the commitTransaction function, so we can't get other more information  about block, `blockResult` packaged in resultLoop() maybe mutch batter!